File: python/difference-of-squares/difference_of_squares.py
def square_of_sum(number):
    return sum(range(1, number + 1)) ** 2


def sum_of_squares(number):
    return sum(map(lambda n: n * n, range(1, number + 1)))


def difference_of_squares(number):
    return square_of_sum(number) - sum_of_squares(number)

File: ruby/difference-of-squares/difference_of_squares.rb
class Squares
  attr_reader :square_of_sum, :sum_of_squares, :difference
  def initialize(number)
    seq = (1..number)
    @square_of_sum = seq.sum.pow(2)
    @sum_of_squares = seq.map { |n| n * n }.sum
    @difference = square_of_sum - sum_of_squares
  end
end
